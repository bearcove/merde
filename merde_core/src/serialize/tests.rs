use crate::{Event, IntoStatic, Map, Serializer, Value};
use insta::assert_debug_snapshot;

#[test]
fn test_serialize() {
    #[derive(Default, Debug)]
    struct ToySerializer {
        events: Vec<Event<'static>>,
    }

    impl Serializer for ToySerializer {
        type Error = ();

        fn write(
            &mut self,
            ev: Event<'_>,
        ) -> impl std::future::Future<Output = Result<(), Self::Error>> {
            self.events.push(ev.into_static());
            async { Ok(()) }
        }
    }

    let mut s = ToySerializer::default();
    let value: Value = Map::new().with("foo", Value::from(42)).into();
    s.serialize_sync(&value).unwrap();

    assert_debug_snapshot!(s.events);
}
