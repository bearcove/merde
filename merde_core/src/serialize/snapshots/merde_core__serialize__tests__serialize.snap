---
source: merde_core/src/serialize/tests.rs
expression: s.events
---
[
    MapStart(
        MapStart {
            size_hint: Some(
                1,
            ),
        },
    ),
    Str(
        "foo",
    ),
    I64(
        42,
    ),
    MapEnd,
]
